### PR TITLE
[new][deps] Include react-with-styles@4.0.0 in peer deps

### PR DESCRIPTION
**Pull request by noratarano**
### Summary

Include `react-with-styles@4.0.0` in peer deps: 
https://github.com/airbnb/react-with-styles/releases/tag/v4.0.0

### Reviewers

@ljharb @TaeKimJR @ahuth @indiesquidge @joeuy @majapw @lencioni

### CODE REVIEW

**Comment by noratarano** [comment]
@ljharb Can this be merged before v4 of RWS is published?

**Comment by noratarano** [comment]
RWS version 4 has been published, so we can merge :)

File: package.json [hunk 1]
```diff
@@ -66,10 +66,10 @@
   },
   "dependencies": {
     "object.assign": "^4.1.0",
-    "react-with-styles-interface-aphrodite": "^5.0.0"
+    "react-with-styles-interface-aphrodite": "^6.0.0"
```
Comment (noratarano) [inline package.json]: This major version includes `react-with-styles@4` as a peer dep as well.

Comment (ljharb) [reply package.json]: if the interface requires 4, not 3 || 4, then this is a major change, and the below peer dep should be just 4 as well

Comment (noratarano) [reply package.json]: @ljharb The interface has 3 || 4. 🤔 So this should be good.

Comment (noratarano) [reply package.json]: It was a breaking change because of the drop of node < 6

**Pull request merged**